src/NUnitFramework/framework/Attributes/RetryAttribute.cs

```diff
@@ -47,16 +59,29 @@ public TestCommand Wrap(TestCommand command)
         public class RetryCommand : DelegatingTestCommand
         {
             private readonly int _tryCount;
+            private readonly Type[] _retryExceptions;
 
             /// <summary>
             /// Initializes a new instance of the <see cref="RetryCommand"/> class.
             /// </summary>
             /// <param name="innerCommand">The inner command.</param>
-            /// <param name="tryCount">The maximum number of repetitions</param>
+            /// <param name="tryCount">The maximum number of repetitions.</param>
             public RetryCommand(TestCommand innerCommand, int tryCount)
+                : this(innerCommand, tryCount, Array.Empty<Type>())
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="RetryCommand"/> class.
+            /// </summary>
+            /// <param name="innerCommand">The inner command.</param>
+            /// <param name="tryCount">The maximum number of repetitions.</param>
+            /// <param name="retryExceptions">A list of exception types, that trigger a retry when thrown.</param>
```

Comment:
Suggest "... that trigger a retry when thrown, in addition to the normal behavior of retrying only on an assertion failure." or similar.